main.go

```diff
@@ -142,7 +161,21 @@ func main() {
 		RHOBSSecretName:              fmt.Sprintf("%v-prom-remote-write", addonName),
 		RHOBSEndpoint:                envVars[rhobsEndpointEnvVarName],
 		RHSSOTokenEndpoint:           envVars[rhssoTokenEndpointEnvVarName],
-	}).SetupWithManager(mgr, nil); err != nil {
+		NetworkType:                  controllers.SDNNetworkType,
+	}
+
+	err = reconciler.Client.Get(ctx, client.ObjectKeyFromObject(egressNetworkPolicyCRD), egressNetworkPolicyCRD)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reconciler.NetworkType = controllers.OVNNetworkType
+		} else {
+			setupLog.Error(err, "Error Getting CRD for EgressNetworkPolicy")
+			os.Exit(1)
+		}
+	}
+
```

Comment:
Why are we introducing this logic between the creation of the controller and `SetupWithManager`?
Please move this logic before the creation of the controller.

Also, it is not recommended to use the reconciler/mgr client for this operation as it will introduce the CRD type into the client cache, and we already know that the client implementation has an issue with cache size.

I would suggest having a util function (in this file) of the form `getNetworkInterfaceType() NetworkInterfaceType` which initializes  a non-cached client which is based on a very minimal schema (you only need the CRD schema) and use it to populate the new the controller field.

Reply:
Done